padertorch/models/dual_path_rnn_source_separator.py

```diff
@@ -0,0 +1,278 @@
+        processed = rearrange(processed, 'b l n -> b n l')
+
+        processed = self.output_proj(self.output_prelu(processed))
+
+        # Split a part of the output for an additional output
+        if self.additional_out_size > 0:
+            processed, additional_out = (
+                processed[..., self.additional_out_size:, :],
+                processed[..., :self.additional_out_size, :][0]
+            )
+
+        # Shape KxBxNxL
+        processed = torch.stack(
+            torch.chunk(processed, self.num_speakers, dim=1))
+
+        processed = self.mask(self.output_nonlinearity(processed), encoded_raw)
+
+        # Decode stream for each speaker
+        decoded = rearrange(
+            self.decoder(rearrange(processed, 'k b n l -> (k b) n l')),
+            '(k b) t -> k b t',
+            k=processed.shape[0], b=processed.shape[1])
+
+        # The length can be slightly longer than the input length if it is not
+        # a multiple of a segment length.
+        decoded = decoded[..., :sequence.shape[-1]]
+
+        # This is necessary if an offset-invariant loss fn (e.g.,
+        # SI-SNR from the TasNet paper) but an offset-variant evaluation metric
+        # (e.g., SI-SDR) is used.
+        # TODO: Fix the loss fn and remove this
+        decoded = decoded - torch.mean(decoded, dim=-1, keepdim=True)
+
+        out = {
+            'out': rearrange(decoded, 'k b t -> b k t'),
+            'encoded': rearrange(encoded_raw, 'b n l -> b l n'),
+            'encoded_out': rearrange(processed, 'k b n l -> b k l n'),
+            'encoded_sequence_lengths': encoded_sequence_lengths,
+        }
+
+        if self.additional_out_size > 0:
+            additional_out = additional_out[..., :processed.shape[-1]]
+            out['additional_out'] = additional_out
+
+        return out
+
+    def loss(self, inputs: dict, outputs: dict) -> dict:
+        s = inputs['s']
+        sequence_lengths = inputs['num_samples']
+        x = outputs['out']
+
+        loss_functions = {
+            'si-sdr': si_sdr_loss,
+            'log-mse': log_mse_loss,
+            'si-sdr-grad-stop': partial(si_sdr_loss, grad_stop=True),
+        }
+        losses = {k: [] for k in loss_functions.keys()}
+
+        for seq_len, estimated, target in zip(sequence_lengths, x, s):
+            for k, loss_fn in loss_functions.items():
+                losses[k].append(
+                    pt.ops.loss.pit_loss(
+                        estimated[:seq_len],
+                        target[:seq_len],
+                        axis=0, loss_fn=loss_fn,
+                    )
+                )
+
+        return {k: torch.mean(torch.stack(v)) for k, v in losses.items()}
+
+    def review(self, inputs, outputs):
+        # Report audios
+        audios = {
+            'observation': pt.summary.audio(
+                signal=inputs['y'][0], sampling_rate=8000
+            ),
+        }
+
```

Comment:
pt.summary.audio does not exist, probably you did not commit your change to the __init__.py